include/aspect/geometry_model/ellipsoidal_chunk.h

```diff
@@ -0,0 +1,254 @@
+
+        Point<3>
+        rotate_paralellegram(const Point<3> &x) const;
+
+      private:
+
+        static double    R;
+        static double    ellipticity;
+
+        Point<3> push_forward_ellipsoid (const Point<3> &phi_theta_d) const;
+        Point<3> pull_back_ellipsoid (const Point<3> &x) const;
+
+        Point<3> push_forward_topo (const Point<3> &phi_theta_d_hat) const;
+        Point<3> pull_back_topo (const Point<3> &phi_theta_d) const;
+    };
+
+  }
+
+}
+namespace aspect
+{
+  namespace GeometryModel
+  {
+    using namespace dealii;
+
+    /**
+     * A class that describes a box geometry of certain width, height, and
+     * depth (in 3d).
+     */
+    /**
+     * A class that describes a geometry for the region below the East African Rift.
+     */
+    template <int dim>
+    class EllipsoidalChunk : public Interface<dim>
+    {
+      public:
+        /**
+         * Generate a coarse mesh for the geometry described by this class.
+         */
+        virtual
+        void
+        create_coarse_mesh(parallel::distributed::Triangulation<dim> &coarse_grid) const;
+
+        /**
+         * Return the typical length scale one would expect of features in this geometry,
+         * assuming realistic parameters.
+         *
+         * We return 1/20th of the distance from the two endpoints (vertex_0 and vertex_7) of the cell.
+         */
+        virtual
+        double
+        length_scale() const;
+
+        virtual
+        double
+        depth(const Point<dim> &position) const;
+
+        virtual Point<dim>
+        representative_point(const double depth) const;
+
+        virtual
+        double
+        maximal_depth() const;
+
+        /**
+         * Return the set of boundary indicators that are used by this model. This
+         * information is used to determine what boundary indicators can be used in
+         * the input file.
+         *
+         * The box model uses boundary indicators zero through 2*dim-1, with the first
+         * two being the faces perpendicular to the x-axis, the next two perpendicular
+         * to the y-axis, etc.
+         */
+        virtual std::set<types::boundary_id>
+        get_used_boundary_indicators() const;
+
+        /*
+        *Set symbolic names for boudaries (mrtf)
+        */
+        virtual std::map<std::string,types::boundary_id>
+        get_symbolic_boundary_names_map() const;
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters(ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter
+         * file.
+         */
+        virtual
+        void
+        parse_parameters(ParameterHandler &prm);
+
+        /**
+         * calculate radius at current position
+         */
+        static
+        double
+        get_radius(const Point<dim> &point);
```

Comment:
Are this and the following function used outside this class? If not, could they be made private?

For the documentation, please start comments with a capital letter and terminate with a period. These are just regular sentences.